src/helpers/prf_compute/multiplicative_inverse.rs

```diff
@@ -0,0 +1,84 @@
+use num_bigint::{BigInt, BigUint, ToBigInt};
+use num_traits::{identities::Zero, One, Signed};
+
+/// Implementation of Extended Euclidean Algorithm to compute multiplicative inverse of a number
+/// i.e given num and modulus it computes t such that (num * t) % modulus = 1
+///
+/// How it computes :
+/// Finds Bezout Coefficient a and b for coprimes num and modulus i.e.
+/// ``bezout_coefficient_a`` * ``num`` + ``bezout_coefficient_b`` * ``modulus`` = 1
+/// This deduces to ( ``bezout_coefficient_a`` * ``num`` ) % ``modulus`` = 1 % ``modulus``
+/// Hence ``bezout_coefficient_a`` % ``modulus`` is the multiplicative inverse of ``num``
+///
+/// Before returning number is converted into a positive number
+///
+/// # Panics
+///
+/// This may return None when converting from f32 or f64,
+/// and will always succeed when converting from any integer or unsigned primitive, or ``BigUint``.
+#[must_use]
+pub fn mod_inverse(num: &BigUint, modulus: &BigUint) -> Option<BigInt> {
+    /// This computes the greatest common divisor (``gcd``) of integers ``num1`` and ``num2``,
+    /// and also the coefficients of Bézout's identity, which are integers ``bezout_coefficient_a`` and ``bezout_coefficient_b`` such that
+    /// ``num1`` * ``bezout_coefficient_a`` + ``num2`` * ``bezout_coefficient_b`` = ``gcd``(num1,num2)
+    #[inline]
+    fn extended_binary_gcd(num1: &BigUint, num2: &BigUint) -> (BigInt, BigUint) {
+        // Assume non zero
+        assert!(!num1.is_zero());
+        assert!(!num2.is_zero());
+
+        let mut old_gcd = num1.clone();
+        let mut gcd = num2.clone();
+
+        let mut bezout_coefficient_old_a = BigInt::one();
+        let mut bezout_coefficient_a = BigInt::zero();
+
+        loop {
+            if old_gcd < gcd {
+                gcd -= &old_gcd;
+                bezout_coefficient_a -= &bezout_coefficient_old_a;
+            } else {
+                old_gcd -= &gcd;
+                bezout_coefficient_old_a -= &bezout_coefficient_a;
+            }
+
+            if old_gcd.is_zero() {
+                return (bezout_coefficient_a, gcd);
+            }
+        }
+    }
+    let (mut bezout_coefficient_a, gcd) = extended_binary_gcd(num, modulus);
+
+    if !gcd.is_one() {
+        return None;
+    }
+
+    let to_add = modulus.to_bigint().unwrap();
+
+    if bezout_coefficient_a.is_negative() {
+        bezout_coefficient_a += &to_add;
+        assert!(bezout_coefficient_a.is_positive());
+        Some(bezout_coefficient_a)
+    } else {
+        Some(bezout_coefficient_a % to_add)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I would recommend adding a few more tests.

Perhaps a good approach would be to hard-code a list of known large primes: (just grab a bunch of largeish ones from this page: https://en.wikipedia.org/wiki/List_of_prime_numbers) and for each prime, make a loop which generates numbers that are strictly less than that prime. Compute the mod_inverse and check that indeed that value is correct by multiplying the two numbers and checking the result is 1 mod P.